test/object-store/sync/session/progress_notifications.cpp

```diff
@@ -395,90 +383,302 @@ TEST_CASE("progress notification", "[sync][session][progress]") {
+            CHECK(transferrable == current_transferrable);
+            CHECK(current_estimate == current_estimate);
+        }
+
+        SECTION("for download notifications") {
+            // Prime the progress updater
+            current_transferred = 60;
+            current_transferrable = 501;
+            double current_estimate = current_transferred / double(current_transferrable);
+            const uint64_t original_transferrable = current_transferrable;
+            progress.update(current_transferred, current_transferrable, 21, 26, 1, current_estimate, 21 / double(26),
+                            0);
+
+            progress.register_callback(
+                [&](auto xferred, auto xferable, double ep) {
+                    transferred = xferred;
+                    transferrable = xferable;
+                    download_estimate = ep;
+                    callback_was_called = true;
+                },
+                NotifierType::download, false, 0);
+            // Wait for the initial callback.
+            REQUIRE(callback_was_called);
+
+            // Now manually call the notifier handler a few times.
+            callback_was_called = false;
+            current_transferred = 66;
+            current_transferrable = 582;
+            current_estimate = current_transferred / double(current_transferrable);
+            progress.update(current_transferred, current_transferrable, 25, 26, 1, current_estimate, 25 / double(26),
+                            0);
+            CHECK(callback_was_called);
+            CHECK(transferred == current_transferred);
+            CHECK(transferrable == original_transferrable);
+            CHECK(download_estimate == current_estimate);
+
+            // Second callback
+            callback_was_called = false;
+            current_transferred = original_transferrable + 100;
+            current_transferrable = 1021;
+            current_estimate = current_transferred / double(current_transferrable);
+            progress.update(current_transferred, current_transferrable, 68, 191, 1, current_estimate,
+                            68 / double(191), 0);
+            CHECK(callback_was_called);
+            CHECK(transferred == current_transferred);
+            CHECK(transferrable == original_transferrable);
+            CHECK(download_estimate == current_estimate);
+
+            // The notifier should be unregistered at this point, and not fire.
+            callback_was_called = false;
+            current_transferred = original_transferrable + 250;
+            current_transferrable = 1228;
+            current_estimate = current_transferred / double(current_transferrable);
+            progress.update(current_transferred, current_transferrable, 199, 591, 1, current_estimate,
+                            199 / double(591), 0);
+            CHECK(!callback_was_called);
+        }
+
+// The functionality of this test was moved out of the object-store and into the sync client.
```

Comment:
I'm not sure what this comment means?

Reply:
it means that we've removed the fake "server_version" parameter to the object-store progress callback and now the sync::Session itself ignores progress updates before it receives its first download message https://github.com/realm/realm-core/blob/c967ba13990ed52bf79ef649353c79a858fffbdf/src/realm/sync/client.cpp#L1871. we can probably remove this test and/or replace it with an integration test?

Reply:
i've verified that we updated the `sync::Session` upload/download progress tests to so we don't get any progress updates until we've received at least one download message, and removed this test in a later commit.

Reply:
Not getting any upload progress updates until we've received at least one download message is itself a breaking change that should be reverted. I think all of the changes to test_sync.cpp in #7124 that aren't just adapting to the changed signature of the progress notification callback are bogus and are resulting in some tests (such as Sync_UploadDownloadProgress_3) not testing the scenario they were specifically written to test.

Reply:
Hm, maybe that's actually wrong and we never were sending upload notifications before the first download. The sync progress notification tests are definitely nonsense now, but it's sort of hard to tell if there's also functional changes.

Reply:
Yeah, this is something I went back and forth in the PR about during the initial implementation. I think there is a functional change in the sync::Session behavior, but it is just behavior that used to be in object-store. You used to get an initial upload notification when you first registered your callback, but we ignored progress updates from the sync::Session until after you'd received the first download.

Reply:
I can take a look at `Sync_UploadDownloadProgress_3` and any other tests to try to make sure they're doing what they say they do, but probably not before this PR is merged.